src/activities/jumbled_words/jumbled_words.js

```diff
@@ -0,0 +1,185 @@
+                + defaultLevelsFile + ". Can't continue!");
+            // any way to error-exit here?
+            return;
+        }
+    }
+}
+
+
+function stop() {
+}
+
+function shuffleString(s)
+{
+    var a = s.split("");
+    var n = a.length;
+
+    for(var i = n-1; i>0; i--) {
+        var j = Math.floor(Math.random() * (i + 1));
+        var tmp = a[i];
+        a[i] = a[j];
+        a[j] = tmp;
+    }
+    return a.join("");
+}
+
+function initLevel() {
+    items.bar.level = currentLevel + 1
+    items.hint.text = "hint";
+
+    if (currentSubLevel == 0) {
+            level = levels[currentLevel];
+            maxSubLevel = level.questions.length;
+            items.score.numberOfSubLevels = maxSubLevel;
+            items.score.currentSubLevel = "1";
+            answers = level.questions[Math.floor(Math.random()*maxSubLevel)];
+            var copy = answers;
+        var demo = shuffleString(answers);
+        if(demo == answers)
+        {
+            demo = shuffleString(answers);
+        }
+        else
+            questions = demo;
+    } else {
+        generate = level.questions[Math.floor((Math.random()*maxSubLevel))];
+        while(copy==generate)
+        {
+            generate = level.questions[Math.floor((Math.random()*maxSubLevel))];
+        }
+        copy=generate;
+        answers=generate;
+         demo = shuffleString(answers);
+        if(demo == answers)
+        {
+            demo = shuffleString(answers);
+        }
+        else
+            questions = demo;
+        items.score.currentSubLevel = currentSubLevel + 1;
+        }
+    items.questionItem.visible = true
+    items.questionItem.text = questions;
+
+}
+
+function validateLevels(levels)
+{
+    var i;
+    for (i = 0; i < levels.length; i++) {
+        if (undefined === levels[i].questions
+            || typeof levels[i].questions != "string"
+            || levels[i].questions.length < 1
+            || typeof levels[i].answers != "string"
+            || levels[i].answers.length < 1)
+            return false;
+    }
+    if (i < 1)
+        return false;
+    return true;
+}
+
+function nextLevel() {
+    if(maxLevel <= ++currentLevel ) {
+        currentLevel = 0
+    }
+    currentSubLevel = 0;
+    initLevel();
+}
+
+function previousLevel() {
+    if(--currentLevel < 0) {
+        currentLevel = maxLevel - 1
+    }
+    currentSubLevel = 0;
+    initLevel();
+}
+
+function nextSubLevel() {
+    if( ++currentSubLevel >= maxSubLevel) {
+        currentSubLevel = 0
+        nextLevel()
+    }
+    initLevel();
+}
+
+function checkAnswer()
+{
+    var modelEntry = items.edit.getText(0, items.edit.text.length)
+    var value = modelEntry.replace(/\s\s*$/, '') //to remove leading white spaces
+    value = modelEntry.replace(/^\s\s*/, '') //to remove trailing white spaces
+    if (value === answers) {
+        items.bonus.good("flower");
+        items.edit.text = "";
```

Comment:
you should avoid setting variables in js because this breaks the qml binding. Can you do it in qml with a condition?
something like, GCText { id: edit; text: condition?qsTr("text1") : "" ... }